dmarc/s3.py

```diff
@@ -420,23 +517,48 @@ def process_payload(self, content_type, payload):
                             credentials = boto3.Session().get_credentials()
                             awsauth = AWS4Auth(credentials.access_key,
                                                credentials.secret_key,
                                                self.es_region,
                                                'es',
                                                session_token=credentials.token)
+
+                            # Check if the index exists and create it
+                            # if necessary
+                            index_only_url = '{}/{}'.format(self.es_url,
+                                                            self.es_index)
+                            response = requests.head(index_only_url,
+                                                     auth=awsauth,
+                                                     timeout=Parser.__Timeout)
+                            if response.status_code != 200:
+                                logging.info('The index {} does not exist.  Creating it.'.format(self.es_index))
+                                try:
+                                    response = requests.put(index_only_url,
+                                                            auth=awsauth,
+                                                            json=Parser.__IndexPayload,
+                                                            headers={'Content-Type': 'application/json'},
+                                                            timeout=Parser.__Timeout)
+                                    # Raises an exception if we didn't get back a
+                                    # 200 code
+                                    response.raise_for_status()
+                                except requests.exceptions.RequestException:
+                                    logging.exception('Unable to create the index {}.'.format(self.es_index))
+                                    return False
+
+                            # Now save the report
+                            full_url = '{}/_doc'.format(index_only_url)
                             try:
-                                response = requests.post(self.es_url,
+                                response = requests.post(full_url,
                                                          auth=awsauth,
                                                          json=jsn,
                                                          headers={'Content-Type': 'application/json'},
                                                          timeout=Parser.__Timeout)
                                 # Raises an exception if we didn't get back a
                                 # 200 code
                                 response.raise_for_status()
-                            except requests.exceptions.RequestException as e:
+                            except requests.exceptions.RequestException:
```

Comment:
Is there any way to get some text output from this RequestException?

Reply:
[`logging.exception`](https://docs.python.org/3/library/logging.html#logging.exception) does exactly that.

Reply:
Got it- I see that now, thanks!